lsp-mode.el

```diff
@@ -3706,73 +3706,94 @@ and the position respectively."
 
 (defun lsp-completion-at-point ()
   "Get lsp completions."
-  (let* ((bounds (bounds-of-thing-at-point 'word-strictly))
-         (trigger-chars (->> (lsp--server-capabilities)
-                             (gethash "completionProvider")
-                             (gethash "triggerCharacters")))
-         start-point result done?)
-    (list
-     (if bounds (car bounds) (point))
-     (point)
-     (lambda (_probe _pred action)
-       (cond
-        ((eq action 'metadata)
-         `(metadata . ((display-sort-function . (lambda (candidates)
-                                                  (--sort (string-lessp (get-text-property 0 'lsp-sort-text it)
-                                                                        (get-text-property 0 'lsp-sort-text other))
-                                                          candidates)))
-                       (category . lsp-substring))))
-        ((eq (car-safe action) 'boundaries) nil)
-        (t (if done?
-               result
-             (setf start-point (point))
-             (let* ((resp (lsp-request-while-no-input "textDocument/completion"
-                                                      (plist-put (lsp--text-document-position-params)
-                                                                 :context (ht ("triggerKind" 1)))))
-                    (items (lsp--sort-completions (cond
-                                                   ((seqp resp) resp)
-                                                   ((hash-table-p resp) (gethash "items" resp))))))
-               (setf done? (or (seqp resp)
-                               (not (gethash "isIncomplete" resp)))
-                     result (seq-into (seq-map (-lambda ((item &as &hash "label"
-                                                               "insertText" insert-text
-                                                               "sortText" sort-text))
-                                                 (propertize (or label insert-text)
-                                                             'lsp-completion-item item
-                                                             'lsp-sort-text sort-text))
-                                               items)
-                                      'list)))))))
-
-     :annotation-function #'lsp--annotate
-     :company-require-match 'never
-     :company-prefix-length (point)
-     :exit-function
-     (lambda (candidate _status)
-       (-let [(&hash "insertText" insert-text
-                     "textEdit" text-edit
-                     "insertTextFormat" insert-text-format
-                     "additionalTextEdits" additional-text-edits)
-              (lsp--resolve-completion (plist-get (text-properties-at 0 candidate) 'lsp-completion-item))]
+  (when (or (--some (lsp--client-completion-in-comments? (lsp--workspace-client it))
```

Comment:
Did you happen to find a server which does not want completion in comments and strings? Because I want to change the default.

Reply:
I think most of the server don't want completion in comment. Inside string, having language aware completion is fine.

Reply:
leave it as it is but we should definitely revisit the logic and investigate what vscode is doing in this case.